### ISSUE TITLE
Issues with WPF?

### ISSUE THREAD

**Issue by jamesc-skyward** [VsixCommunity/Community.VisualStudio.SourceGenerators#4]
I have some Visual Studio extensions currently using WinForms for their UI. I was looking at changing them over to use WPF, but simply adding a WPF control to my project prevents it from building. I get errors that say `The name 'VSCommandTable' does not exist in the current context` and `The name 'Vsix' does not exist in the current context`. I tried some workarounds I found on GitHub, but they weren't working for me. I did come across this message where @reduckted seems to mention a similar issue: https://github.com/madskristensen/MarkdownEditor2022/pull/10#issuecomment-1009824340

Is this a known issue? Or is there something I need to do to get the source generators working alongside WPF?

**Comment by jamesc-skyward** [comment]
Huh. It seems to work fine if I add a window using the [Extensibility Template Pack](https://github.com/VsixCommunity/ExtensibilityTemplatePack).

**Comment by reduckted** [comment]
I can reproduce this, but adding a window from the Extensibility Template Pack didn't fix it. Maybe there was something else that you changed in the project file?

I'm going to reopen this and see if we can find a solution.

**Comment by jamesc-skyward** [comment]
> I can reproduce this, but adding a window from the Extensibility Template Pack didn't fix it. Maybe there was something else that you changed in the project file?
> 
> I'm going to reopen this and see if we can find a solution.

Nope. I changed nothing else. I just have an existing project I'm working with. When I add a new item, if I select "User Control (WPF)" it won't build. I can select "Async Tool Window (Community)" or "Dialog Window (Community)" and everything works fine. No other changes.

I played around with this a bit and I found a way to fix the issue, for me at least. The xaml file that gets created when adding a new WPF User Control has the line `xmlns:local="clr-namespace:{Namespace.To.Xaml}"`. If I remove that line (VS tells me it's unnecessary), everything builds as expected.

**Comment by jamesc-skyward** [comment]
Oh. I should clarify that adding a window with the Extensibility Template Pack doesn't fix things the "bad" one. I'm deleting the "bad" one and then adding a new one from the Extensibility Template Pack. If I have both side-by-side it won't compile.

**Comment by jamesc-skyward** [comment]
So after avoiding using `clr-namespace` in my projects, I started looking into this a bit more. I found https://github.com/reduckted/ProjectFilter where the source generator and WPF are working together. Looking at the differences between my project and that one, I found it has something to do with the .csproj file. The ProjectFilter extension is using the newer SDK csproj format, and when I converted mine over it started working. So either there's something in the old .csproj breaking things that I wiped out in the conversion, or something in the new format itself is fixing things.
